File: src/types/activity.ts
export enum ActivityType {
  DIGITAL = 'DIGITAL',
  NON_DIGITAL = 'NON_DIGITAL'
}

export interface Activity {
  id: number;
  title: string;
  description: string;
  type: ActivityType;
  imageUrl?: string;
  resourceUrl?: string;
}

export interface CreateActivityRequest {
  title: string;
  description: string;
  type: ActivityType;
  imageUrl?: string;
  resourceUrl?: string;
}

File: main/java/com/github/cawtoz/enfokids/model/activity/Activity.java
package com.github.cawtoz.enfokids.model.activity;

import com.github.cawtoz.enfokids.model.activity.enums.ActivityTypeEnum;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "activities")
@Data
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 100, nullable = false)
    private String title;

    @Column(length = 3000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(length = 30, nullable = false)
    private ActivityTypeEnum type;

    private String imageUrl;

    private String resourceUrl;

}

